certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -191,11 +191,9 @@ def deploy_cert(self, domain, cert_path, key_path,
                         vhost.filep, vhost.names)
         except errors.MisconfigurationError as error:
             logger.debug(error)
-            logger.warning(
-                "Cannot find a cert or key directive in %s for %s. "
-                "VirtualHost was not modified.", vhost.filep, vhost.names)
             # Presumably break here so that the virtualhost is not modified
-            return False
+            raise errors.PluginError("Cannot find a cert or key directive in {0} for {1}. "
```

Comment:
There's no unit test coverage here. I think this is valuable, but I'm fine delaying this as long as one one of us test these lines manually.